### ISSUE TITLE
Unpack[TypedDict] incorrectly accepts undeclared keyword arguments for open TypedDicts

### ISSUE THREAD

**Issue by DaBestXD** [astral-sh/ty#4212]
### Summary

Possibly related:  #3757

`ty` incorrectly treats `Unpack[TypedDict]` for an open `TypedDict` as accepting arbitrary extra keyword arguments, whereas the typing spec says only declared keys should be accepted unless `extra_items` is explicitly specified.

Playground link: https://play.ty.dev/8261df74-17cd-411d-bb32-6954576e04cc

```python
from typing import TypedDict, Unpack


class Args(TypedDict):
    x: int


def foo(**kwargs: Unpack[Args]) -> None:
    bar(**kwargs)


def bar(x: int) -> None:
    pass

# Both mypy and pyright report an unexpected parameter
foo(x=1, unexpected="hello")
```

Running `uv run ty check foo.py` reports `All checks passed!` while both mypy and pyright report unexpected parameter found. 

Python: 3.14
Ty: 0.0.69

### Version

0.0.69

**Comment by carljm** [comment]
Thanks for the report!

This behavior is intentional. It is consistent with the semantics of an open TypedDict, and it was intentionally allowed in the typing conformance suite in https://github.com/python/typing/pull/2272 -- more rationale is present in the discussion on that PR. The typing spec text should have been updated in that same PR to not suggest the stricter behavior is required, but it looks like that was missed.

If you want the stricter unpack validation, use a `closed=True` TypedDict.

**Comment by DaBestXD** [comment]
So just to clarify for the updated typing spec for TypedDicts is that all TypedDicts are open unless specified by `closed=True` ?

**Comment by carljm** [comment]
Yes -- that wasn't changed in https://github.com/python/typing/pull/2272 . It has always been true (since `TypedDict` was originally introduced) that `TypedDict` was open: allowed  additional unknown keys to be present at runtime. The subtyping behavior -- allowing a `TypedDict` with extra fields to be a subtype of one with fewer fields -- has always implied that interpretation, too. But in practice type checkers (inconsistently and unsoundly) implemented a few behaviors as if open TypedDicts were actually closed (because often closed TypedDicts are actually more useful).

The PEP that introduced closed TypedDicts made it explicit that the default is open. Arguably if we could do it all over again, we'd make closed the default instead, but it's too late for that.

**Comment by DaBestXD** [comment]
Ok thanks for the help, have a nice weekend! 🐱
